pkg/cloudprovider/providers/openstack/openstack_volumes.go

```diff
@@ -368,6 +373,17 @@ func (os *OpenStack) ExpandVolume(volumeID string, oldSize resource.Quantity, ne
 		return oldSize, err
 	}
 
+	if volume.Status != volumeAvailableStatus {
+		if _, ok := volumes.(*VolumesV3); !ok {
+			// cinder volume can not be expanded if cinder API is not v3
+			return oldSize, fmt.Errorf("volume status is not available")
+		}
+		if volume.Status != volumeInUseStatus {
```

Comment:
why you don't check at line 376?

Reply:
On 376 we're checking the volumeAvailableStatus status. If it is available, then it doesn't matter whether it is V2 or V3. If the status is not available, then we're checking two things:
* if it is not V3, then fail
* if it is V3, then the status must be volumeInUseStatus